stdlib/2and3/socket.pyi

```diff
@@ -505,6 +505,10 @@ class socket:
         def __init__(self, family: int = ..., type: int = ...,
                      proto: int = ..., fileno: Optional[int] = ...) -> None: ...
 
+    if sys.version_info >= (3, 2):
+        def __enter__(self) -> socket: ...
```

Comment:
Strictly speaking this should be `def __enter__(self: _SelfT) -> _SelfT: ...`, where `_SelfT` is a TypeVar bound to socket. This way, the right type will be inferred if somebody uses `with` on a subclass of socket. Not sure how common it is to subclass `socket.socket`, but better to be safe.